File: 1894-merge-strings-alternately/merge-strings-alternately.cpp
class Solution {
public:
    string mergeAlternately(string word1, string word2) {
        int m = word1.size();
        int n = word2.size();
        string ans = "";
        for(int i=0;i<max(m,n);i++){
            if(i<m) ans.push_back(word1[i]);
            if(i<n) ans.push_back(word2[i]);
        }
        return ans; 
    }
};

File: 1894-merge-strings-alternately/merge-strings-alternately.java
class Solution {
    public String mergeAlternately(String word1, String word2) {
        StringBuilder sb = new StringBuilder();
        int i=0;
        int j=0;
        while(i<word1.length() || j<word2.length()){
            if(i<word1.length()) sb.append(word1.charAt(i++));
            if(j<word2.length()) sb.append(word2.charAt(j++));
        }
        return sb.toString();
    }
}